### ISSUE TITLE
CLI does not upload a perfect-fidelity copy of artifacts

### ISSUE THREAD

**Issue by nebhale** [cloudfoundry/cli#676]
As seen in https://github.com/cloudfoundry/java-buildpack/issues/244, it appears that the CLI does not upload a perfect copy of artifacts during `push`.  Given a WAR file with the following contents:

```
META-INF/MANIFEST.MF
META-INF/
WEB-INF/
WEB-INF/classes/
WEB-INF/lib/
index.jsp
```

a buildpack is presented with the following filesystem:

```
META-INF/
META-INF/MANIFEST.MF
index.jsp
```

If I had to guess, the CLI is not replicating empty directories.  Since the existence of the `WEB-INF/` directory is the only thing that makes a WAR file Servlet spec compliant, that directory must be on the filesystem at staging time, empty or not.  While this exhibits for a select set of Java WAR applications, I'd definitely have a look to see where else the CLI isn't creating a perfect-fidelity copy of artifacts.

/cc @jbayer @dkoper

**Comment by nebhale** [comment]
@dkoper I've verified that this now works properly with `cf version 6.13.0+0331291-2015-11-17`.

**Comment by dkoper** [comment]
Thanks Ben!

**Comment by dkoper** [comment]
FYI, this fix is included in today's 6.14.0 CLI release.
https://github.com/cloudfoundry/cli#downloads

Regards,
Dies Koper
CF CLI PM
